[Yingying] ip

yyingg-243 opened the pull request.

yyingg-243 committed:
Give the chatbot a new name

implement the skeletal version

yyingg-243 committed:
echo what the user inputted

add line and indentation to make it looks nice

able to echo what the user input

yyingg-243 committed:
add array to store user's input and print it out

add an array with a fixed size 100

read in user's input and save to array

able to list out the array

able to exit

yyingg-243 committed:
add the ability to mark as done

add a new class which is task

able to mark and unmark task

convert the string array to task array

yyingg-243 committed:
remove unnecessary code

replace some attributes as constant

edit the settings to adhere coding standard

tejaskumar0 commented on src/main/java/ChattyDuke.java:
```diff
@@ -0,0 +1,83 @@
+        System.out.println();
+
+        Scanner scanner = new Scanner(System.in);
+        Boolean isBye = false;
```
Good naming convention for boolean variables

tejaskumar0 commented on src/main/java/ChattyDuke.java:
```diff
@@ -0,0 +1,83 @@
+
+    public static void main(String[] args) {
+
+        Task[] items = new Task[MAX_ITEMS];
```
plural form used to name arrays, good coding standard

tejaskumar0 commented on src/main/java/ChattyDuke.java:
```diff
@@ -0,0 +1,83 @@
+import java.util.Scanner;
+
+public class ChattyDuke {
```
maybe can include more comments to make the flow clearer

tejaskumar0 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+public class Task {
+    protected String description;
```
good use of protected fields

Yikbing commented on src/main/java/ChattyDuke.java:
```diff
@@ -0,0 +1,83 @@
+
+            if(s1.equalsIgnoreCase("bye")) {
+                isBye = true;
+                System.out.println(INDENTATION + "Bye. Hope to see you again soon!");
```
extraction can be done in the bye statement :D

Yikbing commented on src/main/java/ChattyDuke.java:
```diff
@@ -0,0 +1,83 @@
+        Scanner scanner = new Scanner(System.in);
+        Boolean isBye = false;
+
+        while(!isBye) {
```
can make the happy path prominent by restructuring the code such that the isBye is nested instead of the "usual" case(which is when the user does not type "bye")

Yikbing commented on src/main/java/ChattyDuke.java:
```diff
@@ -0,0 +1,83 @@
+                isBye = true;
+                System.out.println(INDENTATION + "Bye. Hope to see you again soon!");
+                System.out.println(INDENTATION + LINE_SEPARATOR);
+            } else if (s1.equalsIgnoreCase("list")) {
```
good job for using the else-if statements, avoiding the arrowhead style code!

Yikbing commented on src/main/java/ChattyDuke.java:
```diff
@@ -0,0 +1,83 @@
+public class ChattyDuke {
+
+    public static final String LINE_SEPARATOR = "____________________________________________________________";
+    public static final int MAX_ITEMS = 100;
```
good job for explaining the meaning of the 100 and not having "magic numbers" in your code! :D

Yikbing reviewed (commented):
Overall, well done on the code! You just need a little bit of extraction of code to be able to let readers know what the code does! :D

yyingg-243 committed:
code able to handle 3 type of task

added a new class taskManager

create methods to handle list, mark, unmark logic

yyingg-243 committed:
edit code to follow coding standard

yyingg-243 committed:
Convert the if else to switch statement

yyingg-243 committed:
Add exception handling

Able to handle error gracefully

add chattyDukeException class

edit code to make it more readable

TVageesan commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,213 @@
+            if(taskInfo.length < 2 || taskInfo[1].trim().isEmpty()){
+                throw new ChattyDukeException("Hmmmm, Please include the task description");
+
+            }else{
```
We could reduce nesting here by removing the else. If the condition passes and the error throws, the rest of the code won't execute anyway, so the else is not needed. This is a good example of using the inversion pattern to reduce nesting in general!

TVageesan commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,213 @@
+                    break;
+
+                case EVENT:
+                    eventTask(s1, taskLists);
```
The method names 'todoTask', 'deadlineTask', etc. are a bit vague. They do not clearly describe the function's actual purpose - to create a new task object. A general rule of thumb is to have a action/verb in your method names: i.e. `createTodo`, `deleteTodo` etc.

TVageesan commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,213 @@
+
+
+
+            taskLists[ChattyDuke.inputCount] = new Event(firstPart, fromDescription, toDescription);
```
ChattyDuke's static variable 'inputCount' seems to be keeping track of the number of tasks in your taskList. If the only usage of this variable is within TaskManager, it might make more sense for it to be a instance variable stored only inside TaskManager.

TVageesan commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,213 @@
+
+        String taskInfo = s1.split(" ", 2)[1];
+        taskLists[ChattyDuke.inputCount] = new Task(taskInfo);
+        Task t = taskLists[ChattyDuke.inputCount];
```
Avoid abbreviatiating variable names that you use multiple places in your code. I promise the 3 characters you save here isn't worth the readability loss for people reading it being used later: `t.markDone()` vs `task.markDone()`.

TVageesan commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,213 @@
+            String description = s1.split(" ",2)[1];
+            String[] taskInfo = description.split("/from", 2);
+
+            String firstPart = taskInfo[0].trim();
+            String secondPart = taskInfo[1].trim();
```
'firstPart', 'secondPart' are not very descriptive variable names. It would be more useful if they reflected their contents, i.e. 'description', 'time'. I noticed this in multiple places.

TVageesan reviewed (commented):
Well done logic wise, but you could reconsider some of the variable/method names for better readability!

yyingg-243 committed:
Add support for deleting tasks from the list

Change the array to arraylist

yyingg-243 committed:
Save the tasks in the hard disk

yyingg-243 committed:
Merge branch 'branch-Level-6'

yyingg-243 committed:
fix code due to aborting the merge confilct

yyingg-243 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/ChattyDuke.java
#	src/main/java/TaskManager.java

yyingg-243 committed:
edit code to load the saved data

add the jar file

yyingg-243 committed:
edit code to load saved data

yyingg-243 committed:
Fix bug of saving data into files

enable the code to handle if folder not exits

yyingg-243 committed:
Make the code more OOP
Split code in ChattyDuke file to other file
Ui, Storage, Parser file are created

yyingg-243 committed:
delete unwanted files

yyingg-243 committed:
Give users a way to find task via keyword

use "Fine" command to search

Add a new method in taskManager class

Edit code in Storage class to reload previous task in arraylist

yyingg-243 committed:
update readme file

edit indentation format in the code

yyingg-243 committed:
include javadoc comments

yyingg-243 committed:
edit storage file to save data properly

edit javadoc

yyingg-243 committed:
Merge pull request #1 from yyingg-243/branch-checking

edit javadoc and storage file
